### PR TITLE
Adds query to Retraction Watch database.

### PR DESCRIPTION
**Pull request by crsh**
This PR adds optional queries to the Retraction Watch database. Because there is currently no API for the database, the queries are performed by submitting the HTTP form and scraping the returned HTML. The server responses are currently rather slow and the queries take some time.

There are at least two things to note and discuss:

1. Due to the slow queries, I thought about querying the Retraction Watch database only if openretractions.com does not return an update (in case both are queried). This behavior could be made optional by adding an additional parameter such as `return = "all"` vs. `return = "unique"`.
2. The Retraction Watch database does not return the title of the update and hence, `query_rw()` returns `NA` in the respective column. The title could additionally be fetched from CrossRef using `rcrossref` but that seemed unnecessary to me.

What do you think?

### CODE REVIEW

File: R/retractcheck.R [hunk 1]
```diff
@@ -198,3 +204,139 @@ query_or <- function(doi) {
...
+      }
+
+      res$update_type <- ifelse(grepl('Retraction', rw_update_table$Article.Type.s.Nature.of.Notice), 'retraction', res$update_type)
+      res$update_type <- ifelse(is.na(res$update_type) & grepl('Expression of concern', rw_update_table$Article.Type.s.Nature.of.Notice), 'expression of concern', res$update_type)
+      res$update_type <- ifelse(is.na(res$update_type) & grepl('Correction', rw_update_table$Article.Type.s.Nature.of.Notice), 'correction', res$update_type)
+
+      res$retracted <- ifelse(res$update_type == 'retraction', TRUE, FALSE)
+      res$update_doi <- find_doi(rw_update_table$Retraction.or.Other.NoticesDate.PubMedID.DOI)
+
+      # if("rcrossref" %in% installed.packages()) {
```
Comment (chartgerink) [inline R/retractcheck.R:324]: Is this necessary to retain?

**Review by chartgerink — APPROVED** [review]
Thanks for the updates! Will merge.

Could you let me know whether the commented out `rcrossref` part can be deleted?

**Comment by crsh** [comment]
Thanks. The `rcrossref` part can go if you feel like `NA` for the title of the update is fine (see description of PR above). Any thoughts on the suggestion above to speed up the database queries?

**Comment by chartgerink** [comment]
Oh oops oversaw that when responding. I think it's best to minimize dependencies so to not include it. I'll leave the commented code in for now in case usage deems it preferable to include it anyway.

**Pull request merged**